app/components/primer/avatar_component.rb

```diff
@@ -34,17 +34,25 @@ def initialize(src:, alt:, size: 20, square: false, href: nil, **system_argument
       @system_arguments[:height] = size
       @system_arguments[:width] = size
 
-      @system_arguments[:classes] = class_names(
-        system_arguments[:classes],
-        "avatar" => !href,
+      avatar_class_names = class_names(
```

Comment:
I think we can keep  `system_arguments[:classes]` here, so If an user wants to add a custom class, it adds the class to the `<a>` instead of `<img>`.
At least it looks like that'd be the correct behavior since we are adding all classes to `<a>`

Reply:
I wondered about that. I'll make the change and update the `CHANGELOG`. Thanks for the quick review! 💛

Reply:
this is roughly what I had in mind

```rb
avatar_class_names = class_names(
  "avatar",
  system_arguments[:classes],
  "avatar--small" => size < SMALL_THRESHOLD,
  "circle" => !square
)

if @href
  @link_arguments = { href: href, classes: avatar_class_names }
  @system_arguments[:classes] = nil # otherwhise it will apply the custom classes twice
else
  @system_arguments[:classes] = avatar_class_names
end
```

Reply:
if we go this route, I'd ❤️  to see a test showing the custom class behavior too! :)